block/src/Pos/Block/Logic/Header.hs

```diff
@@ -244,6 +247,9 @@ classifyHeaders inRecovery headers = do
                           depthDiff blkSecurityParam
             | otherwise -> CHsValid lcaChild
 
+
+data GetHeadersFromManyToError = GHFBadRequest Text deriving (Show,Generic)
```

Comment:
Minor nit: networking language. `GHFBadInput`, `GHFUnknownHeader` or similar instead? Sure, this is *used* by the diffusion layer / networking but it's intended to be modular.

Reply:
Yes, fixed. It wasn't intentional, `BadInput` is better.